clustering_ranges_walker.hh

```diff
@@ -69,13 +74,18 @@ private:
             if (!_current_range) {
                 _current_start = position_in_partition_view::before_all_clustered_rows();
             } else {
-                _current_start = position_in_partition_view::for_range_start(_current_range.front());
-                _current_end = position_in_partition_view::for_range_end(_current_range.front());
+                if (_schema.current_transformation() == schema_transformation_type::reversed) {
+                    _current_start = position_in_partition_view::for_range_reverse_start(_current_range.front());
+                    _current_end = position_in_partition_view::for_range_reverse_end(_current_range.front());
+                } else {
+                    _current_start = position_in_partition_view::for_range_start(_current_range.front());
+                    _current_end = position_in_partition_view::for_range_end(_current_range.front());
+                }
```

Comment:
I think we'll need some words to explain things, because this is confusing. IIUC _current_range is reversed, and you're un-reversing it.

Let's call the clustering keys as passed in from the query as in the "query domain", and the unreversed keys and comparators the "sstable domain". Without reversed queries, they are the same. With reversed queries, they aren't. We can start by adding comments about what domain a function input or output or variable is in, and later perhaps change the variable and function names too. I wouldn't go as far as to use the type system, but maybe it's merited.

Reply:
The confusion around `position_in_partition_view` mostly arises from the fact, that when using the reversed schema, the comparisons on clustering key prefixes are reversed, but the comparisons on `bound_weight` remain the same. 
We treat a key as included in a range if it's not less than the beginning of the range, but it is less than the end of the range. As a result, if we had a clustering range including keys from ck1 to ck2, we would have a beginning position_in_partition_view looking like (ck1, before), and an ending like (ck2, after). If we want to reverse this range so that the comparisons work with the reversed schema, we can't simply swap the beginning and the end (because the comparisons on "before" and "after" weren't reversed), but use a start (ck2, before) and (ck1, after) instead.

In this case `_current_range` is actually in the "query domain" and we're simply setting correct `bound_weight`s for the reversed range, using the end key as a start, and the start key as end.

Reply:
(and yes, potentially reversed == query domain, unreversed == sstable/table domain)

Reply:
Ok